tests/PHPStan/Analyser/NodeScopeResolverTest.php

```diff
@@ -447,6 +447,12 @@ public function dataFileAsserts(): iterable
 		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-4970.php');
 		yield from $this->gatherAssertTypes(__DIR__ . '/data/bug-5322.php');
 		yield from $this->gatherAssertTypes(__DIR__ . '/data/splfixedarray-iterator-types.php');
+
+		if (PHP_VERSION_ID >= 80000) {
+			yield from $this->gatherAssertTypes(__DIR__ . '/data/mb_substitute_character-php8.php');
+		} elseif (PHP_VERSION_ID >= 70200) {
```

Comment:
What's the behaviour on PHP 7.1 and can it be tested?

Reply:
The behavior difference on PHP 7.1 is the returned integer range when the function is called without args. On PHP 7.1, the expected range is `int<0, 65534>` vs `int<0, 1114111>` for 7.2+. It can most certainly be tested, but I'm not sure the best way to do that without adding another file just for PHP 7.1.

Reply:
Please just add a special file for PHP 7.1 :)